qiskit/circuit/register.py

```diff
@@ -117,7 +118,7 @@ def __getitem__(self, key):
             CircuitError: if the `key` is not an integer.
             QiskitIndexError: if the `key` is not in the range `(0, self.size)`.
         """
-        if not isinstance(key, (int, slice, list)):
+        if not isinstance(key, (int, slice, list, numbers.Integral)):
```

Comment:
`numbers.Integral` covers `int` already, we don't need both 
```suggestion
        if not isinstance(key, (numbers.Integral, slice, list)):
```